crdschurch/crds-search-index-data
# US18241 Removes entries excluded from Crossroads from Algolia index

## Task

Stop Contentful entries with **Yes** selected under **Exclude from crossroads.net?** from being added to the Algolia index.

[Original Task Link](https://rally1.rallydev.com/#/38108142417d/custom/292180449324?qdp=%2Fdetail%2Fuserstory%2F338799135432)

## Solution

The index already skipped excluded entires as long as **Yes** is selected on the first save. I have introduced a `removeExcludedFromIndex` method in `content.model.js` that will execute once `updateIndexContent` is complete to make sure all items are successfully removed.

## Tester Notes

- Create a new article or use this [example article](https://app.contentful.com/spaces/y3a9myzsdjan/environments/int/entries/5NTWz8cailrxUZ49OB4Smo) to toggle **Exclude from crossroads.net?** on and off. 
- Then send a `POST` requests using the following routes.
  - Refresh all Articles in index with `/content/article`. *You can replace article with any content type*
  - Refresh all content with `content/refreshAll`
  - Update an individual record using `/content/` with a body containing a contentful content change payload. You can view the search webhook history for int [here](https://app.contentful.com/spaces/y3a9myzsdjan/settings/webhooks/7JxoTxlv99MHxfILw5pgvv) to retrieve an example body.
      - **Note:** I am not sure if it's possible to test an individual article when it's saved on int without it being merged in.
- Check the [Algolia index](https://www.algolia.com/apps/8Y3N3H5PNJ/explorer/browse/int_crds) each time by searching using the **ObjectID** instead of the regular **Search**. **Note:** Sometimes the Algolia index search page needed to be refreshed when a record was deleted.

## Preview

- You can use [Postman](https://www.getpostman.com/) or another API client like it to make the `POST` requests
- [Algolia index](https://www.algolia.com/apps/8Y3N3H5PNJ/explorer/browse/int_crds) for Contentful int environment
